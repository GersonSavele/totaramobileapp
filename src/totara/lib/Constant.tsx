/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author Tharaka Dushmantha <tharaka.dushmantha@totaralearning.com>
 */

export const DEVICE_REGISTRATION = "X-TOTARA-MOBILE-DEVICE-REGISTRATION";
export const WEBVIEW_SECRET = "X-TOTARA-MOBILE-WEBVIEW-SECRET";
export const X_API_KEY = "X-API-KEY";

export const DATE_FORMAT = "D, MMM YYYY";

export const PLATFORM_IOS = "ios";
export const PLATFORM_ANDROID = "";

export const NAVIGATION_COURSE_DETAILS ="CourseDetails";
export const NAVIGATION_PROGRAM_DETAILS ="ProgramDetails";
export const NAVIGATION_SETTING = "Settings";

export const SCREEN_WIDTH_X_LARGE = "xlarge";
export const SCREEN_WIDTH_LARGE = "large";
export const SCREEN_WIDTH_MEDIUM = "medium";
export const SCREEN_WIDTH_SMALL = "small";

export const GRADE_INFORMATION_HEADER_TITLE = "Your highest grade"
export const GRADE_INFORMATION_HEADER_BUTTON_TITLE = "Begin"
export const GRADE_INFORMATION_STATUS = "VIEW TO COMPLETE"

export const GRADE_FEEDBACK_HEADER_TITLE = "Your highest grade"
export const GRADE_FEEDBACK_HEADER_BUTTON_TITLE = "Attempt again"
export const GRADE_FEEDBACK_STATUS = "PASSED"




