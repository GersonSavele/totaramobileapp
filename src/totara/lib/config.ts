/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import localConfig from "./config.local";
import ConsoleLogger from "./logger/ConsoleLogger";

const defaultConfig = {
  urlProtocol: "https",
  mobileApi: {
    persistentQuery: true
  },
  userAgent: "TotaraMobileApp",
  apiUri: (host: string) => `${host}/totara/mobile/api.php`,
  infoUri: (host: string) => `${host}/totara/mobile/site_info.php`,
  loginUri: (host: string) => `${host}/login/index.php`,
  findLearningUri: (host: string) => `${host}/totara/catalog/index.php`,
  deviceRegisterUri: (host: string) => `${host}/totara/mobile/device_register.php`,
  webViewUri: (host: string) => `${host}/totara/mobile/device_webview.php`,
  forgotPasswordUri: (host: string) => `${host}/login/forgot_password.php`,
  nativeLoginSetupUri: (host: string) => `${host}/totara/mobile/login_setup.php`,
  nativeLoginUri: (host: string) => `${host}/totara/mobile/login.php`,
  appLinkDomain: "https://mobile.totaralearning.com",
  deepLinkSchema: "totara://",
  minApiVersion: "2020100100",
  rootOfflineScormPlayer: "html",
  sentryUri: "",
  disableConsoleYellowBox: false,
  devOrgUrl: "" // Default organization url only used in dev mode.
};

export const internalConfig = {
  logger: ConsoleLogger
};

let config = { ...defaultConfig, ...localConfig };

export default config;
