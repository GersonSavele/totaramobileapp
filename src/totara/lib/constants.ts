/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

const DEVICE_REGISTRATION = "X-TOTARA-MOBILE-DEVICE-REGISTRATION";
const WEBVIEW_SECRET = "X-TOTARA-MOBILE-WEBVIEW-SECRET";
const AUTHORIZATION = "AUTHORIZATION";
const AUTH_HEADER_FIELD = "x-api-key";

const DATE_FORMAT = "D MMM, YYYY";
const DATE_FORMAT_FULL = "dddd, DD MMMM YYYY, h:mm A";
const SECONDS_FORMAT = "X";

const PLATFORM_IOS = "ios";
const PLATFORM_ANDROID = "android";
const ANDROID_STATUSBAR_HEIGHT = 75;

const SCREEN_WIDTH_X_LARGE = "xlarge";
const SCREEN_WIDTH_LARGE = "large";
const SCREEN_WIDTH_MEDIUM = "medium";
const SCREEN_WIDTH_SMALL = "small";

const FILE_EXTENSION = ".zip";
const OFFLINE_SCORM_PREFIX = "OfflineSCORM_";

enum ActivityModType {
  scorm = "scorm",
  label = "label",
  resource = "resource"
}

const DEFAULT_LANGUAGE = "en";

const PULL_TO_REFRESH_OFFSET = 50;

enum SubPlugin {
  findLearning = "findlearning"
}

export {
  DEVICE_REGISTRATION,
  WEBVIEW_SECRET,
  AUTHORIZATION,
  AUTH_HEADER_FIELD,
  DATE_FORMAT,
  DATE_FORMAT_FULL,
  SECONDS_FORMAT,
  PLATFORM_IOS,
  PLATFORM_ANDROID,
  SCREEN_WIDTH_X_LARGE,
  SCREEN_WIDTH_LARGE,
  SCREEN_WIDTH_MEDIUM,
  SCREEN_WIDTH_SMALL,
  OFFLINE_SCORM_PREFIX,
  FILE_EXTENSION,
  ActivityModType,
  DEFAULT_LANGUAGE,
  ANDROID_STATUSBAR_HEIGHT,
  PULL_TO_REFRESH_OFFSET,
  SubPlugin
};
