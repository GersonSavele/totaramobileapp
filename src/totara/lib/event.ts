/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import EventEmitter from "react-native/Libraries/vendor/emitter/EventEmitter";

const event = new EventEmitter();

const EVENT_LISTENER = "event_listener";

enum Events {
  Logout,
  NetworkError
}

export { Events, EVENT_LISTENER };
export default event;
