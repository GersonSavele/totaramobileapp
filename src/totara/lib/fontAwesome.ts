/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import {
  faHome,
  faCloudDownloadAlt,
  faUser,
  faVideo,
  faTimes,
  faChevronRight,
  faFilm,
  faListUl,
  faTasks,
  faComments,
  faExternalLinkAlt,
  faCheck,
  faChevronUp,
  faChevronDown,
  faExclamationTriangle,
  faLock,
  faBoxOpen,
  faExclamationCircle,
  faArrowRight,
  faArrowLeft,
  faBullhorn,
  faBookOpen,
  faPenSquare,
  faBook,
  faPen,
  faChevronLeft,
  faUnlockAlt,
  faCaretUp,
  faCaretDown,
  faTrashAlt,
  faCaretRight,
  faBolt,
  faCheckCircle,
  faInfoCircle,
  faAngleRight,
  faInfo,
  faBan,
  faPaperclip,
  faPlay,
  faEye,
  faEyeSlash,
  faShareAlt
} from "@fortawesome/free-solid-svg-icons";
import { library } from "@fortawesome/fontawesome-svg-core";

const init = () => {
  library.add(
    faHome,
    faCloudDownloadAlt,
    faUser,
    faVideo,
    faTimes,
    faChevronRight,
    faFilm,
    faListUl,
    faTasks,
    faComments,
    faExternalLinkAlt,
    faCheck,
    faExclamationTriangle,
    faChevronUp,
    faLock,
    faBoxOpen,
    faExclamationCircle,
    faArrowLeft,
    faArrowRight,
    faBullhorn,
    faBookOpen,
    faPenSquare,
    faBook,
    faPen,
    faChevronLeft,
    faUnlockAlt,
    faCaretUp,
    faCaretDown,
    faTrashAlt,
    faCaretRight,
    faBolt,
    faChevronDown,
    faCheckCircle,
    faInfoCircle,
    faAngleRight,
    faInfo,
    faBan,
    faPaperclip,
    faPlay,
    faEyeSlash,
    faEye,
    faShareAlt
  );
};

const FontAwesome = {
  init
};

export default FontAwesome;
