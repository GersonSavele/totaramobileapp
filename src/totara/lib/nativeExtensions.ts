import { NativeModules } from "react-native";
import { isIOS } from "./tools";
const { TotaraExtensions } = NativeModules;

const setNotificationBadgeCount = (count: number) => {
  if (isIOS) {
    TotaraExtensions.setNotificationBadgeCount(count);
  }
};
export { setNotificationBadgeCount };
