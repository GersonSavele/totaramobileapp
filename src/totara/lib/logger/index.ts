/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import { Logger } from "./Logger";
import { internalConfig } from "../config";

const LoggerImpl = internalConfig.logger; // TODO would good to make this injectable, do a simple dependency injection

export { LoggerImpl as Log, Logger };
