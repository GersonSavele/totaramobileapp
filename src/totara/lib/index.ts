/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import config from "./config";
import { Logger, Log } from "./logger";
import { showMessage } from "./tools";
import * as constants from "./constants";

export { config, Logger, Log, showMessage, constants };
