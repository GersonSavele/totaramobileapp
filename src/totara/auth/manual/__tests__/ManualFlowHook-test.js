/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author Jun Yamog <jun.yamog@totaralearning.com>
 */

import { renderHook, act } from "@testing-library/react-hooks";
import {
  ManualFlowSteps,
  useManualFlow,
  manualFlowReducer,
} from "../ManualFlowHook";
import { config } from "@totara/lib";

describe("useManualFlow", () => {
  it("should be on step done when url and secret is valid", async () => {
    const onLoginSuccess = jest.fn(setupSecret => {
      expect(setupSecret.uri).toBe("https://success.com");
      expect(setupSecret.secret).toBe("theSecret");
    });

    const { result, waitForNextUpdate } = renderHook(
      ({ onLoginSuccess }) =>
        useManualFlow(mockFetch)({ onLoginSuccess: onLoginSuccess }),
      { initialProps: { onLoginSuccess: onLoginSuccess } }
    );

    act(() => {
      result.current.onSiteUrlSuccess("https://success.com");
    });

    await act(async () => waitForNextUpdate());

    expect(result.current.manualFlowState).toMatchObject({
      flowStep: ManualFlowSteps.webview,
      isSiteUrlSubmitted: true,
      siteUrl: "https://success.com"
    });

    act(() => {
      result.current.onSetupSecretSuccess("theSecret");
    });

    expect(result.current.manualFlowState).toMatchObject({
      flowStep: ManualFlowSteps.done,
      isSiteUrlSubmitted: false,
      setupSecret: "theSecret",
      siteUrl: "https://success.com"
    });
    expect(onLoginSuccess).toBeCalledTimes(1);
  });

  it("should be on step fail when url is invalid", async () => {
    const mockFetchError = () => Promise.reject(new Error("server error"));
    const { result, waitForNextUpdate } = renderHook(() => useManualFlow(mockFetchError)());
    
    act(() => {
      result.current.onSiteUrlSuccess("https://fail.com");
    });

    expect(result.current.manualFlowState).toMatchObject({
      isSiteUrlSubmitted: true,
      isSiteUrlFailure: false,
      siteUrl: "https://fail.com"
    });

    await act(async () => waitForNextUpdate());

    expect(result.current.manualFlowState).toMatchObject({
      isSiteUrlSubmitted: false,
      isSiteUrlFailure: true,
      siteUrl: "https://fail.com"
    });
  }); 
});

describe("manualFlowReducer", () => {
  it("should put flowStep into siteUrl when it is cancelled", () => {
    const currentState = {
      isSiteUrlSubmitted: true,
      flowStep: ManualFlowSteps.native
    };
    const action = {
      type: "cancelManualFlow"
    };

    const newState = manualFlowReducer(currentState, action);
    expect(newState.flowStep).toBe(ManualFlowSteps.siteUrl);
    expect(newState.isSiteUrlSubmitted).toBeFalsy();
  });

  it("should use the auth on the SiteInfo for the next flowstep", () => {
    const currentState = {
      isSiteUrlSubmitted: true,
      flowStep: ManualFlowSteps.siteUrl
    };
    const testSiteInfo = {
      auth: "native",
      siteMaintenance: false,
      theme: {
        logoUrl: "https://mytotara.client.com/totara/mobile/logo.png",
        colorBrand: "#CCFFCC"
      },
      version: "2019101802"
    };
    const action = {
      type: "apiSuccess",
      payload: testSiteInfo
    };

    const newState = manualFlowReducer(currentState, action);
    expect(newState.flowStep).toBe(ManualFlowSteps.native);
    expect(newState.siteInfo).toMatchObject(testSiteInfo);
  });

  it("should set flowStep to incompatible for unsupport minApiVersion", () => {
    config.minApiVersion = "2029101802";
    const currentState = {
      isSiteUrlSubmitted: true,
      flowStep: ManualFlowSteps.siteUrl
    };
    const testSiteInfo = {
      auth: "native",
      siteMaintenance: false,
      theme: {
        logoUrl: "https://mytotara.client.com/totara/mobile/logo.png",
        colorBrand: "#CCFFCC"
      },
      version: "2019101802"
    };
    const action = {
      type: "apiSuccess",
      payload: testSiteInfo
    };

    const newState = manualFlowReducer(currentState, action);
    expect(newState.flowStep).toBe(ManualFlowSteps.incompatible);
    expect(newState.siteInfo).toMatchObject(testSiteInfo);
  });

  it("should set flowStep to native for disabled app minApiVersion", () => {
    config.minApiVersion = "disabled";
    const currentState = {
      isSiteUrlSubmitted: true,
      flowStep: ManualFlowSteps.siteUrl
    };
    const testSiteInfo = {
      auth: "native",
      siteMaintenance: false,
      theme: {
        logoUrl: "https://mytotara.client.com/totara/mobile/logo.png",
        colorBrand: "#CCFFCC"
      },
      version: "2019101802"
    };
    const action = {
      type: "apiSuccess",
      payload: testSiteInfo
    };

    const newState = manualFlowReducer(currentState, action);
    expect(newState.flowStep).toBe(ManualFlowSteps.native);
    expect(newState.siteInfo).toMatchObject(testSiteInfo);
  });

  it("should properly init the state", () => {
    const currentState = {
      isSiteUrlSubmitted: false,
      siteUrl: undefined,
      flowStep: ManualFlowSteps.siteUrl
    };
    const action = {
      type: "apiInit",
      payload: "https://totarasite.com"
    };

    const newState = manualFlowReducer(currentState, action);
    expect(newState.flowStep).toBe(ManualFlowSteps.siteUrl);
    expect(newState.isSiteUrlSubmitted).toBeTruthy();
    expect(newState.siteUrl).toBe("https://totarasite.com");
  });

  it("should put flowstep as done, set the setupSecret when setupSecretSuccess", () => {
    const currentState = {
      isSiteUrlSubmitted: true,
      siteUrl: "https://totarasite.com",
      flowStep: ManualFlowSteps.native
    };
    const action = {
      type: "setupSecretSuccess",
      payload: "the_secret"
    };

    const newState = manualFlowReducer(currentState, action);
    expect(newState.flowStep).toBe(ManualFlowSteps.done);
    expect(newState.isSiteUrlSubmitted).toBeFalsy();
    expect(newState.setupSecret).toBe("the_secret");
  });

  it("should api failure, when site-info was Submitted", () => {
    const currentState = {
      isSiteUrlSubmitted: true,
      siteUrl: "https://totarasite.com",
      flowStep: ManualFlowSteps.native
    };
    const action = {
      type: "apiFailure",
      payload: new Error()
    };
    const newState = manualFlowReducer(currentState, action);
    expect(newState.flowStep).toBe(ManualFlowSteps.native);
    expect(newState.isSiteUrlSubmitted).toBeFalsy();
    expect(newState.isSiteUrlFailure).toBe(true);
  });
});

const mockFetch = () => {
  return Promise.resolve({
    auth: "webview",
    siteMaintenance: false,
    theme: {
      logoUrl: "https://mytotara.client.com/totara/mobile/logo.png",
      colorBrand: "#CCFFCC"
    },
    version: "2019101802"
  });
};