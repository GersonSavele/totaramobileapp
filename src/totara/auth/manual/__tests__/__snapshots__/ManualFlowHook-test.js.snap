// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manualFlowReducer should set flowStep to browser login and take screen-shot in Information modal 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onDismiss={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.3)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 4,
          "height": "70%",
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 24,
            "marginVertical": 16,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/resources/images/browser_login/browser_login.png",
            }
          }
          style={
            Object {
              "alignItems": "center",
              "resizeMode": "contain",
              "width": 525,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 24,
            "marginVertical": 16,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Object {
              "color": "#3D444B",
              "fontSize": 32,
              "fontWeight": "700",
              "lineHeight": 41,
              "paddingBottom": 16,
              "textAlign": "center",
            }
          }
        >
          translated browser_login.title with undefined
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "color": "#3D444B",
              "fontSize": 17,
              "fontWeight": "400",
              "lineHeight": 22,
              "paddingBottom": 16,
              "textAlign": "center",
            }
          }
        >
          translated browser_login.description with undefined
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "justifyContent": "space-between",
            "marginVertical": 16,
            "minHeight": 104,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              this is a mock button
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              this is a mock button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
